cpp/src/barretenberg/honk/composer/ultra_honk_composer.test.cpp

```diff
@@ -0,0 +1,1015 @@
+    //     auto val_honk = honk_prover.wire_polynomials[0][i];
+    //     // auto val_honk = honk_store.get(label)[i];
+    //     auto val_plonk = plonk_store.get(label)[i];
+    //     if (val_honk != val_plonk) {
+    //         info("UNEQUAL index = ", i);
+    //         info("honk: ",val_honk);
+    //         info("plonk: ", val_plonk);
+    //     }
+    // }
+}
+
+TEST(UltraHonkComposer, create_gates_from_plookup_accumulators)
+{
+    auto honk_composer = UltraHonkComposer();
+    auto plonk_composer = proof_system::plonk::UltraComposer();
+
+    barretenberg::fr input_value = fr::random_element();
+    {
+
+        const fr input_hi = uint256_t(input_value).slice(126, 256);
+        const fr input_lo = uint256_t(input_value).slice(0, 126);
+        const auto input_hi_index = honk_composer.add_variable(input_hi);
+        const auto input_lo_index = honk_composer.add_variable(input_lo);
+
+        const auto sequence_data_hi =
+            plookup::get_lookup_accumulators(plookup::MultiTableId::PEDERSEN_LEFT_HI, input_hi);
+        const auto sequence_data_lo =
+            plookup::get_lookup_accumulators(plookup::MultiTableId::PEDERSEN_LEFT_LO, input_lo);
+
+        const auto lookup_witnesses_hi = honk_composer.create_gates_from_plookup_accumulators(
+            plookup::MultiTableId::PEDERSEN_LEFT_HI, sequence_data_hi, input_hi_index);
+        const auto lookup_witnesses_lo = honk_composer.create_gates_from_plookup_accumulators(
+            plookup::MultiTableId::PEDERSEN_LEFT_LO, sequence_data_lo, input_lo_index);
+    }
+    {
+        const fr input_hi = uint256_t(input_value).slice(126, 256);
+        const fr input_lo = uint256_t(input_value).slice(0, 126);
+        const auto input_hi_index = plonk_composer.add_variable(input_hi);
+        const auto input_lo_index = plonk_composer.add_variable(input_lo);
+
+        const auto sequence_data_hi =
+            plookup::get_lookup_accumulators(plookup::MultiTableId::PEDERSEN_LEFT_HI, input_hi);
+        const auto sequence_data_lo =
+            plookup::get_lookup_accumulators(plookup::MultiTableId::PEDERSEN_LEFT_LO, input_lo);
+
+        const auto lookup_witnesses_hi = plonk_composer.create_gates_from_plookup_accumulators(
+            plookup::MultiTableId::PEDERSEN_LEFT_HI, sequence_data_hi, input_hi_index);
+        const auto lookup_witnesses_lo = plonk_composer.create_gates_from_plookup_accumulators(
+            plookup::MultiTableId::PEDERSEN_LEFT_LO, sequence_data_lo, input_lo_index);
+    }
+
+    auto honk_prover = honk_composer.create_prover();
+    auto plonk_prover = plonk_composer.create_prover();
+
+    verify_consistency(honk_prover, plonk_prover);
+}
+
+/**
+ * @brief Build UltraHonkComposer
+ *
+ */
+TEST(UltraHonkComposer, test_no_lookup_proof)
+{
+    auto honk_composer = UltraHonkComposer();
+    auto plonk_composer = proof_system::plonk::UltraComposer();
+
+    size_t MM = 4;
+    for (size_t i = 0; i < MM; ++i) {
```

Comment:
Maybe there is merit in implementing these as templated lambda functions and then just using two calls?